core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -170,6 +171,7 @@
     private final Function<TableScanNode, TableInfo> tableInfoSupplier;
     private final Map<DynamicFilterId, DynamicFilterDomainStats> dynamicFilterDomainStats;
     private final ValuePrinter valuePrinter;
+    private final boolean verbose;
```

Comment:
Pass `verbose` to `Visitor`'s constructor instead of recording it in `PlanPrinter`.